x/profile/internal/types/profile.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Profile represents a generic account on Desmos, containing the information of a single user
+type Profile struct {
+	Name             string         `json:"name,omitempty"`
+	Surname          string         `json:"surname,omitempty"`
+	Moniker          string         `json:"moniker"`
+	Bio              string         `json:"bio,omitempty"`
```

Comment:
Probably better to have this as `*string` instead.

I would also recommend using a builder pattern for this object. So that if you want to build a new profile you do something like this: 

```golang
NewProfile(moniker).WithName(name).WithSurname(surname)...
```

Each function should be defined as follows (example with name): 

```golang
func (p Profile) WithName(name string) Profile {
  p.Name = name
  return p
}
```